### ISSUE TITLE
New feature : multi insert

### ISSUE THREAD

**Issue by ndrean** [ricardombiot/surreal_ex#1]
Thanks for this one! 

I can nicely insert an Elixir map into a table with `SurrealEx.HTTP.Table.create(map)`. It seems that you reach the REST API. Could you implement a multi-insert? I guess this works for the "/sql" endpoint with `insert into`.

```elixir
list_maps = [%{c: 1}, %{c: 2}]
query = Jason.encode!(list_maps)

url_sql = "http://localhost:8000/sql"
auth = [hackney: [basic_auth: {"root", "root"}]]
headers = [
  {"accept", "application/json"},
  {"Content-Type", "application/json"},
  {"NS", "testns"},
  {"DB", "testdb"}
]

Conn.sql(" insert into my_table #{query};")
```

**Comment by ricardombiot** [comment]
Hi, Elixir friend! 
Of course, would can implement new methods for make this or other operations more accessible for clients but...  

I don't know if would be philosophical right... in terms of design because now, this is an library that makes as an infrastructure layer (HTTP) that covers the official Surreal DBs  HTTP API endpoints.

If we started adding not official methods even if these are simple cover of queries... the maintane of library will suffer.  I am worry include errors that affects the security and stability.

_Solutions:_
- Create a simple helper module on your project that support the additional methods that you needs. This methods would be part of your project maintenance and doesnt depends of me (this library).
- Well... also, you can speaks with Tobie and suggest that open official endpoints for that cases... this library wouldnt be the responsable of its maintenance (security, stability...so on), because would be part of official support. 

In summary, I think that sometimes less is better. While the devs can build from this library any query... fulfills its mission.

**Comment by ricardombiot** [comment]
Note: I know that this answer can be a bit disappointed, because you know that would be more easy for me implemented your request that tried convince you the possible derivate problems of accept include this kind of requests. 

In fact, sure I had loss more time on this answer... that I would lost implement it. I trust you can undestant that I think that should rejected this kind of features while I havent the time for evolve this library to be more that only a infrastructure HTTP API cover.

**Comment by ndrean** [comment]
Humm, you may be right. I was able to do it anyway, no problem, and thank you again for this work.
I understand that we want to reach the REST API provided by SurrealDB. On the contrary, I almost use the "/sql" endpoint as I find this more straightforward, and less verbose. You implemented the excellent behaviour `use SurrealEx.Conn `, then I can use `Conn.sql` for almost any query.  You also included a version with a JWT token which I did not exploit for the moment, so I know that a "secured" query is available.

What I couldn't do is conditional queries using Elixir inputs. An example: below I can wrap this into a function where "my_table" is an argument. This is parsed. However, the "LET" is not parsed if I set an Elixir argument. Don't know how to do it.

```elixir
def my_query("my_table")
LET $status='dev' 
LET $dev_team = (SELECT agency.team[WHERE status=$status] FROM app_table:#{my_table});
SELECT name, status from $dev_team; 
end
```

I am looking forward to the documentation on Websockets. Will you implement it?

I made a [post](https://dev.to/ndrean/testing-surrealdb-1kjl#compare-with-insert-csv-file-with-sqlite) in order not to forget about SurrealDB and SurrealEX!

[![Run in Livebook](https://livebook.dev/badge/v1/blue.svg)](https://livebook.dev/run?url=https%3A%2F%2Fgithub.com%2Fdwyl%2Flearn-elixir-with-livebook%2Fblob%2Fmain%2Fsurrealdb.livemd)
